### PR TITLE
Add flag to disable runtime

### PR DESCRIPTION
**Pull request by azakus**
Only useful for native CSS Custom Properties and native ShadowDOM or
ShadyDOM with `polymer-css-build`

### CODE REVIEW

File: entrypoints/apply-shim.js [hunk 1]
```diff
@@ -211,7 +211,8 @@ if (!window.ShadyCSS || !window.ShadyCSS.ScopingShim) {
 
     nativeCss: nativeCssVariables,
     nativeShadow: nativeShadow,
-    cssBuild: cssBuild
+    cssBuild: cssBuild,
```
Comment (sorvell) [inline entrypoints/apply-shim.js]: Why not just
```
cssBuild,
disableRuntime
```

File: entrypoints/custom-style-interface.js [hunk 2]
```diff
@@ -74,7 +74,8 @@ if (!window.ShadyCSS) {
     flushCustomStyles() {},
     nativeCss: nativeCssVariables,
     nativeShadow: nativeShadow,
-    cssBuild: cssBuild
+    cssBuild: cssBuild,
```
Comment (sorvell) [inline entrypoints/custom-style-interface.js]: Same as previous comment.

File: entrypoints/scoping-shim.js [hunk 3]
```diff
@@ -94,7 +94,9 @@ window.ShadyCSS = {
 
   nativeShadow: nativeShadow,
 
-  cssBuild: cssBuild
+  cssBuild: cssBuild,
```
Comment (sorvell) [inline entrypoints/scoping-shim.js]: Same as previous comment.

File: src/scoping-shim.js [hunk 4]
```diff
@@ -79,7 +79,7 @@ export default class ScopingShim {
    * @param {string=} typeExtension
    */
   prepareTemplateStyles(template, elementName, typeExtension) {
-    if (template._prepared) {
+    if (template._prepared || disableRuntime) {
```
Comment (sorvell) [inline src/scoping-shim.js]: should this also be in `prepareTemplateDom`?

File: tests/disable-runtime.html [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+      test('disableRuntime setting is visible', function() {
+        assert(window.ShadyCSS.disableRuntime, 'disableRuntime setting should be true');
+      })
+      test('no placeholders', function() {
+        const spy = sinon.spy(document, 'createComment');
+        makeElement('x-foo');
+        spy.restore();
+        assert(spy.notCalled, 'placeholder was created but should not have been');
+      });
+    });
```
Comment (sorvell) [inline tests/disable-runtime.html]: Add test that proves no scoping occurs.

File: entrypoints/apply-shim.js (hunk 1 above)
Comment (azakus) [reply entrypoints/apply-shim.js]: Closure Compiler didn't like the shorthand

File: src/scoping-shim.js (hunk 4 above)
Comment (azakus) [reply src/scoping-shim.js]: Will do

File: tests/disable-runtime.html (hunk 5 above)
Comment (azakus) [reply tests/disable-runtime.html]: Will do

File: src/scoping-shim.js (hunk 4 above)
Comment (azakus) [reply src/scoping-shim.js]: Done

File: tests/disable-runtime.html (hunk 5 above)
Comment (azakus) [reply tests/disable-runtime.html]: Done

**Pull request merged**